Framework/API/src/InfoComponentVisitor.cpp

```diff
@@ -36,11 +58,14 @@ void InfoComponentVisitor::registerComponentAssembly(
     child->registerContents(*this);
   }
   const size_t detectorStop = m_assemblySortedDetectorIndices.size();
-
   m_ranges.emplace_back(std::make_pair(detectorStart, detectorStop));
 
   // For any non-detector we extend the m_componetIds from the back
   m_componentIds.emplace_back(assembly.getComponentID());
+
+  m_positions->emplace_back(Kernel::toVector3d(assembly.getPos()));
+  m_rotations->emplace_back(Kernel::toQuaterniond(assembly.getRotation()));
+  clearPositionAndRotationsParameters(m_pmap, assembly);
```

Comment:
This clears positions and rotations immediately. Thus, once you visit the children of the assembly, you will not see their original position and rotation anymore, right? Or am I missing something that takes that into account?